src/app/ConsoleMenu.java

```diff
@@ -0,0 +1,35 @@
+package app;
+
+import app.menu.*;
+
+import java.util.HashMap;
+
+import static app.menu.MenuConstants.*;
+
+public class ConsoleMenu {
+    protected HashMap<Integer, MenuOption> menuItems = new HashMap<>();
+
+    public ConsoleMenu() {
+        menuItems.put(ENCRYPT, new MenuOptionEncrypt());
+        menuItems.put(DECRYPT, new MenuOptionDecrypt());
+        menuItems.put(EXIT_APP, new MenuOptionExit());
+    }
+
+    public void run() {
+        while (isRunning) {
+            System.out.print("\n" + MENU_INFO);
+            String userInput = scanner.nextLine();
+            MenuOption menuItem = menuItems.getOrDefault(enterUserInput(userInput), new MenuOptionIncorrectItem());
+
+            menuItem.runApp();
+        }
+    }
+
+    int enterUserInput(String input) {
+        try {
+            return Integer.parseInt(input);
+        } catch (NumberFormatException e) {
+            return 0;
```

Comment:
why return 0?

Reply:
1. "If I return any numbers like 0, 4, 5 etc. (excluding 1, 2, 3), the program will operate, but when I input certain letters such as 'sdf' or any other non-numeric characters, the program responds: 'Invalid input! Please enter a valid number...', requiring not just valid numbers but also valid alphanumeric characters."

2. The provided Java code is a method that attempts to parse a string input into an integer. If the string cannot be parsed into an integer (for example, if it contains non-numeric characters), the method catches a NumberFormatException and returns 0.

3. The console output shown demonstrates a prompt to the user to input a number corresponding to an operation: "Encrypt Text File - 1", "Decrypt Text File - 2", or "Exit Console - 3". If the user inputs a non-numeric character or a number outside the given options (such as "123" or "sdf"), the console displays an error message: "Invalid input! Please enter a valid number...". The user is then re-prompted for a valid input.

```
Please enter to number:
Encrypt Text File - 1
Decrypt Text File - 2
Exit Console - 3
123
Invalid input! Please enter a valid number...

Please enter to number:
Encrypt Text File - 1
Decrypt Text File - 2
Exit Console - 3
sdf
Invalid input! Please enter a valid number...

Please enter to number:
Encrypt Text File - 1
Decrypt Text File - 2
Exit Console - 3
```